File: Data Structures & Algorithms/valid-binary-search-tree/submission-2.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    void inOrderTraversal(TreeNode* node,vector<int>&iOList) {
        if(node->left!=nullptr) {
            inOrderTraversal(node->left,iOList);
        }
        iOList.push_back(node->val);

        if(node->right!=nullptr) {
            inOrderTraversal(node->right,iOList);
        }
    }

    bool isValidBST(TreeNode* root) {
        if(root==nullptr)return true;
        vector<int>iOList;
        inOrderTraversal(root,iOList);

        for(int i=0;i<iOList.size();i++) {
            cout<<iOList[i]<<" ";
        }

        for(int i=0;i<iOList.size()-1;i++) {
            if(iOList[i]>=iOList[i+1])return false;
        }    
        return true;
    }
};

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        in_order_traversal=[]
        def inOrderTraversal(node):
            nonlocal in_order_traversal
            if node.left:
                inOrderTraversal(node.left)
            
            in_order_traversal.append(node.val)

            if node.right:
                inOrderTraversal(node.right)

        inOrderTraversal(root)
        return in_order_traversal[k-1]